process/block/preprocess/transactions.go

```diff
@@ -424,8 +433,27 @@ func (txs *transactions) CreateAndProcessMiniBlock(sndShardId, dstShardId uint32
 	strCache := process.ShardCacherIdentifier(sndShardId, dstShardId)
 	txStore := txs.txPool.ShardDataStore(strCache)
 
+	var (
+		err             error
+		orderedTxs      []*transaction.Transaction
+		orderedTxHashes [][]byte
+	)
+
 	timeBefore := time.Now()
-	orderedTxes, orderedTxHashes, err := SortTxByNonce(txStore)
+
+	txs.mutOrderedTxs.RLock()
```

Comment:
this could be moved inside a separate method. It simply returns the orderedTxs, first time it orders, second time it just copies.